addons/website_sale/__init__.py

```diff
@@ -1,7 +1,17 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
-
+from odoo import api, SUPERUSER_ID
 from . import controllers
 from . import models
 from . import wizard
 from . import report
+
+
+def uninstall_hook(cr, registry):
+    ''' Need to reenable the `product` pricelist multi-company rule that were
+        disabled to be 'overriden' for multi-website purpose
+    '''
+    env = api.Environment(cr, SUPERUSER_ID, {})
+    multi_company_rules = env.ref('product.product_pricelist_comp_rule', raise_if_not_found=False)
```

Comment:
This will fail if one the the rules can't be found:
```
In [3]: multi_company_rules = self.env.ref('product.product_pricelist_comp_rule', raise_if_not_found=False)

In [4]: multi_company_rules
Out[4]: ir.rule(32,)

In [5]: multi_company_rules += self.env.ref('product.product_pricelist_comp_ruleXXXX', raise_if_not_found=False)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-5-c4fe14dd8fc2> in <module>()
----> 1 multi_company_rules += self.env.ref('product.product_pricelist_comp_ruleXXXX', raise_if_not_found=False)

/home/nim/src/odoo/odoo/models.py in __add__(self, other)
   4674     def __add__(self, other):
   4675         """ Return the concatenation of two recordsets. """
-> 4676         return self.concat(other)
   4677 
   4678     def concat(self, *args):

/home/nim/src/odoo/odoo/models.py in concat(self, *args)
   4683         for arg in args:
   4684             if not (isinstance(arg, BaseModel) and arg._name == self._name):
-> 4685                 raise TypeError("Mixing apples and oranges: %s.concat(%s)" % (self, arg))
   4686             ids.extend(arg._ids)
   4687         return self.browse(ids)

TypeError: Mixing apples and oranges: ir.rule(32,).concat(None)

```

Reply:
Well spotted, I added `or env['ir.rule']` in case of not found, if the user deleted himself the records.